agent/consul/leader.go

```diff
@@ -586,6 +586,53 @@ func (s *Server) initializeManagementToken(name, secretID string) error {
 	return nil
 }
 
+func (s *Server) upsertManagementToken(name, secretID string) error {
```

Comment:
can this be reused inside `initializeManagementToken` at all since it duplicates logic there?

Reply:
yes, I think I can do that. Just wanted to implement the approach and make sure it made sense first. I'll try to de-duplicate some of the logic